src/libbson/src/bson/bson-json.c

```diff
@@ -103,7 +103,6 @@ static const char *read_state_names[] = {FOREACH_READ_STATE (GENERATE_STRING)};
    BS (DECIMAL128)                                                   \
    BS (DBPOINTER)                                                    \
    BS (SYMBOL)                                                       \
-   BS (DBREF)                                                        \
```

Comment:
I believe this removes all of the DBRef-specific code in the extended JSON parser, but please let me know if I missed anything.

Note: this does not affect `$dbPointer` parsing, which also happens to use a subset of the DBRef structure (required `$ref` string and `$id` as an ObjectID with no other keys). Those values are parsed as a normal embedded document and all validation is deferred to `_bson_json_read_append_dbpointer`.